include/em4x70.h

```diff
@@ -13,19 +13,28 @@
 //
 // See LICENSE.txt for the text of the license.
 //-----------------------------------------------------------------------------
-// Low frequency EM4x70 structs
+// Low frequency EM4x70 structs -- common to both ARM firmware and client
 //-----------------------------------------------------------------------------
 
 #ifndef EM4X70_H__
 #define EM4X70_H__
 
+#include <stdint.h>
+#include <stdbool.h>
+
 #define EM4X70_NUM_BLOCKS 16
 
 // Common word/block addresses
 #define EM4X70_PIN_WORD_LOWER 10
 #define EM4X70_PIN_WORD_UPPER 11
 
 typedef struct {
+    // ISSUE: `bool` type does not have a standard-defined size.
+    //        therefore, compatibility between architectures / 
+    //        compilers is not guaranteed.
+    // ISSUE: C99 has no _Static_assert() ... was added in C11
+    // TODO: add _Static_assert(sizeof(bool)==1);
+    // TODO: add _Static_assert(sizeof(em4x70_data_t)==36);
     bool parity;
```

Comment:
The code is compiled with `-Wpedantic`, which seems to suggest pure standards compliance is a goal.   The standard does not provide a size for the `bool` type.  Therefore, anytime `bool` is used in a structure that is used across two or more platforms, it should throw a warning.   It's not a warning under `-Wpedantic`, but likely because the compiler is not aware of which structures are sent "across the wire".

Here, I chose not to make any change, but instead just document a potential `gotcha` because this codebase is mature.  Proper RPC (with byte ordering swapping) is highly recommended for next-generation codebase, of course.

Alternatively, if we can shift to compiling the code as C11, this can simply become a static assertion.  That would allow the unwritten assumption that `sizeof(bool)==1` to error out at compile time if ever not true.

Are you OK leaving the comment in?

Reply:
The comment is fine,

This is how we declare structs for arm <-> client communications.  

```
typedef struct {
    uint32_t magic;
    uint16_t length : 15;  // length of the variable part, 0 if none.
    bool ng : 1;
    uint16_t cmd;
} PACKED PacketCommandNGPreamble;
```

Where you see we use the PACKED and also define the bool to be one bit.